tests/system/small/test_empty_index.py

```diff
@@ -0,0 +1,168 @@
+import pandas as pd
+import pytest
+
+import bigframes.exceptions
+import bigframes.pandas as bpd
+
+
+def test_empty_index_materialize(
+    scalars_df_empty_index, scalars_pandas_df_default_index
+):
+    bf_result = scalars_df_empty_index.to_pandas()
+    pd.testing.assert_frame_equal(
```

Comment:
presumably we'll need to make this robust to non-total ordering at some point, right?

Reply:
We do have a variant that ignore ordering completely. Once we bring in unordered mode, we will want to validate partial orderings, which we don't have a utility for yet.